### PR TITLE
Jenkins

### PR DESCRIPTION
**Pull request by IYermakov**

### CODE REVIEW

File: ops/cloudformation/jenkins.yml [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+                  - ec2:AssociateAddress
+                  - ec2:CreateSecurityGroup
+                  - ec2:DescribeSecurityGroups
+                  - ec2:AuthorizeSecurityGroupIngress
+                  - ec2:DescribeAccountAttributes
+                  - ec2:DeleteSecurityGroup
+                  - secretsmanager:GetSecretValue
+                  - cloudwatch:PutMetricAlarm
+                  - cloudwatch:DeleteAlarm
+                  - rds:CreateDBSubnetGroup
```
Comment (mike-unitskyi) [inline ops/cloudformation/jenkins.yml]: do you really need these permissions?

File: ops/cloudformation/jenkins.yml [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    Type: String
+    Description: The S3 path
+    Default: devopsa3
+
+  SSLCertificate:
+    Type: String
+    Default: 'arn:aws:acm:us-east-1:054017840000:certificate/5c2ab5ef-ea79-419f-9596-9b570a4a7838'
+    Description: ARN of Certificate attached to ELB
+
+  DNSName:
```
Comment (mike-unitskyi) [inline ops/cloudformation/jenkins.yml]: would add one more parameter `DNSSuffix` with `""` Default value for test purpose.

File: ops/cloudformation/jenkins.yml [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    Default: ci
+
+  R53HostedZone:
+    Type: String
+    Default: 'devopsa3.me.uk.'
+    Description: Base DNS domain name
+
+Mappings:
+  RegionMap:
+    us-east-1:
```
Comment (mike-unitskyi) [inline ops/cloudformation/jenkins.yml]: would add more regions

File: ops/cloudformation/install.sh [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+}
+
+#-------------------------------------------------------------------------------
+# Set hostname and perform any necessary related tasks.
+#
+# @param $1 - The desired hostname of the server.
+#-------------------------------------------------------------------------------
+function set_hostname() {
+  hostnamectl set-hostname "${1}"
+# AMI:
```
Comment (mike-unitskyi) [inline ops/cloudformation/install.sh]: remove comments

File: ops/cloudformation/jenkins.yml [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+      InstanceType: !Ref InstanceType
+      AssociatePublicIpAddress: True
+      UserData:
+        Fn::Base64: !Sub |
+          #!/bin/bash
+          set -o errexit
+          set -o nounset
+          set -o xtrace
+
+          aws s3 cp "s3://${S3Path}/scripts/jenkins/install.sh" - | bash -l -s -- --hostname jenkins
```
Comment (mike-unitskyi) [inline ops/cloudformation/jenkins.yml]: it makes sense to add versioning(additional CFN parameter) for script.
also, add healthcheck verification and cfn-signal handling.

File: ops/cloudformation/jenkins.yml (hunk 2 above)
Comment (IYermakov) [reply ops/cloudformation/jenkins.yml]: Didn't catch what you mean.

File: ops/cloudformation/jenkins.yml (hunk 1 above)
Comment (IYermakov) [reply ops/cloudformation/jenkins.yml]: I'm sure jenkins will pretend on more permissions. Do you mean I should add all of them later and use a clear instance profile now?

**Pull request closed without merging**